packages/flutter/lib/src/services/raw_keyboard_web.dart

```diff
@@ -0,0 +1,191 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import 'keyboard_key.dart';
+import 'keyboard_maps.dart';
+import 'raw_keyboard.dart';
+
+/// Platform-specific key event data for Web.
+///
+/// See also:
+///
+///  * [RawKeyboard], which uses this interface to expose key data.
+class RawKeyEventDataWeb extends RawKeyEventData {
+  /// Creates a key event data structure specific for Web.
+  ///
+  /// The [keyCode] and [metaState] arguments must not be null.
+  const RawKeyEventDataWeb({
+    this.keyCode = 0,
+    this.keyLabel,
+    this.metaState = 0,
+  })  : assert(keyCode != null),
+        assert(metaState != null);
+
+  /// The key code corresponding to this key event.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode>
+  /// for more information.
+  final int keyCode;
+
+  /// The modifiers that were present when the key event occurred.
+  ///
+  /// See `lib/src/engine/keyboard.dart` in the web engine for the numerical
+  /// values of the `metaState`. These constants are also replicated as static
+  /// constants in this class.
+  ///
+  /// See also:
+  ///
+  ///  * [modifiersPressed], which returns a Map of currently pressed modifiers
+  ///    and their keyboard side.
+  ///  * [isModifierPressed], to see if a specific modifier is pressed.
+  ///  * [isControlPressed], to see if a CTRL key is pressed.
+  ///  * [isShiftPressed], to see if a SHIFT key is pressed.
+  ///  * [isAltPressed], to see if an ALT key is pressed.
+  ///  * [isMetaPressed], to see if a META key is pressed.
+  final int metaState;
+
+  @override
+  final String keyLabel;
+
+  @override
+  PhysicalKeyboardKey get physicalKey {
+    if (kWebToPhysicalKey.containsKey(keyCode)) {
+      return kWebToPhysicalKey[keyCode] ?? PhysicalKeyboardKey.none;
```

Comment:
Is it possible to have a key with a `null` value? If not, then this one line is sufficient as a complete implementation of this getter.